homeassistant/components/telegram_bot/__init__.py

```diff
@@ -450,21 +450,47 @@ def _render_template_attr(data, attribute):
     return True
 
 
-def initialize_bot(p_config):
+def initialize_bot(p_config: dict, hass: HomeAssistant) -> Bot:
     """Initialize telegram bot with proxy support."""
     api_key = p_config.get(CONF_API_KEY)
     proxy_url = p_config.get(CONF_PROXY_URL)
     proxy_params = p_config.get(CONF_PROXY_PARAMS)
 
     if proxy_url is not None:
-        # These have been kept for backwards compatibility, they can actually be stuffed into the URL.
-        # Side note: In the future we should deprecate these and raise a repair issue if we find them here.
-        auth = proxy_params.pop("username"), proxy_params.pop("password")
+        auth = None
+        if proxy_params is None:
+            # CONF_PROXY_PARAMS has been kept for backwards compatibility.
+            proxy_params = {}
+        elif "username" in proxy_params and "password" in proxy_params:
+            # Auth can actually be stuffed into the URL, but the docs have previously
+            # indicated to put them here.
+            auth = proxy_params.pop("username"), proxy_params.pop("password")
+            ir.async_create_issue(
+                hass,
+                DOMAIN,
+                "proxy_params_auth_deprecation",
+                breaks_in_ha_version="2024.6.0",
```

Comment:
?

Reply:
Nowadays we normally do six months from the estimated release of the deprecation.

Reply:
```suggestion
                breaks_in_ha_version="2024.10.0",
```